File: ios/TileGame/DataObjects/TilePosition.swift
//
//  TilePosition.swift
//  TileGame
//
//  Created by MCC on 15/4/18.
//  Copyright © 2018 Tejesh Alimilli. All rights reserved.
//

import Foundation

struct TilePosition: Equatable, CustomStringConvertible {
    var row = 0
    var column = 0
    
    init(row: Int, column: Int) {
        self.row = row
        self.column = column
    }
    
    init(indexPath: IndexPath) {
        row = indexPath.section
        column = indexPath.row
    }
    
    var indexPath: IndexPath {
        return IndexPath(row: column, section: row)
    }
    
    var description: String {
        return "row \(row) column \(column)"
    }
}

File: android/app/src/main/java/com/alimilli/android/tilegame/dataObjects/TilePosition.kt
package com.alimilli.android.tilegame.dataObjects

data class TilePosition(var row: Int = 0, var column: Int = 0) {
    fun equals(other: TilePosition?): Boolean {
        if (other == null) {
            return false
        }
        return row == other.row && column == other.column
    }

    override fun toString(): String {
        return "row $row column $column"
    }
}
